frontend/src/store/atoms/singlePost.ts

```diff
@@ -0,0 +1,18 @@
+import axios from "axios";
+import { atomFamily, selectorFamily } from "recoil";
+import { BACKEND_URL } from "../../config";
+
+export const singlePost = atomFamily({
+    key:"SinglePost",
+    default:selectorFamily({
+        key:"singlePostSelector",
+        get:(id)=>async()=>{
+            const res = await axios.get(`${BACKEND_URL}/api/v1/blog/${String(id)}`,{
+                headers:{
+                    "Authorization":`Bearer ${localStorage.getItem("token")}`
+                }
+            });
+            return res.data.post;
```

Comment:
Please help me understand the logic here. 

User requests for a post, the singlePost atom makes an API call and returns the blog data.
When user requests for the same post again, how are we ensuring that we don't make the API call again and return the cached data instead ?

Reply:
it is atomFamily so there is multiple atom in one atomFamily,by using these we ensure that different id make different atom and same id atom already exits so it returns that state only

Reply:
atomaFamily checks cache, If the id is already cached, it returns the cached value; otherwise, it will refetch  the data. we can see that if we click on that same post again there is no loading..

Reply:
@aadeshkulkarni can u help me this conflicts?